### PR TITLE
Two coverity fixes

### PR DESCRIPTION
**Pull request by martinpitt**
RHEL's Coverity scanner got fixed to get along with glib ref counting, so the scan results are down to just two unit test issues. These are both false positives, but let's address them to have a clean slate.

https://bugzilla.redhat.com/show_bug.cgi?id=1938693

### CODE REVIEW

File: src/tls/test-server.c [hunk 1]
```diff
@@ -375,6 +375,7 @@ teardown (TestCase *tc, gconstpointer data)
   g_assert_cmpint (wait (NULL), ==, -1);
   g_assert_cmpint (errno, ==, ECHILD);
   /* connection should fail */
+  /* coverity[leaked_handle : FALSE] */
   g_assert_cmpint (do_connect (tc), ==, -ECONNREFUSED);
```
Comment (allisonkarlitskaya) [inline src/tls/test-server.c:379]: Annoying that it still doesn't understand this pattern :(

**Review by allisonkarlitskaya — APPROVED** [review]
Thanks!

File: src/tls/test-server.c (hunk 1 above)
Comment (martinpitt) [reply src/tls/test-server.c:379]: I'm not really sure how to rewrite that in a different way. Just storing the result in a variable and then asserting it doesn't fundamentally change the data flow, I suppose it just doesn't know about the intention of test-* code not making sense with assertions being compiled out?

Comment (kdudka) [reply src/tls/test-server.c:379]: Yes, it is annoying.  I tried to extend Coverity's model of `connect()` to set `errno` in the failing case but it did not make this false positive go away.

Comment (kdudka) [reply src/tls/test-server.c:379]: We already model `g_assertion_message()` as `__coverity_panic__()`.  The problem here seems to be that Coverity fails to infer that `fd` is closed if and only if `do_connect()` returns a negative value, which `-ECONNREFUSED` is for sure.

**Comment by martinpitt** [comment]
@allisonkarlitskaya : Is there some way to restart LGTM? I did log into the website, but I didn't find a retry button. Perhaps you have one, as you registered it?

**Comment by allisonkarlitskaya** [comment]
I don't have this button :(

**Comment by martinpitt** [comment]
OK, just landing then.

**Pull request merged**